cmd/boulder/main_test.go

```diff
@@ -9,6 +9,35 @@ import (
 	"github.com/letsencrypt/boulder/test"
 )
 
+func TestFailedConfigValidation(t *testing.T) {
+	type FooConfig struct {
+		VitalValue       string `validate:"required"`
+		VoluntarilyVoid  string
+		VisciouslyVetted string `validate:"omitempty,endswith=baz"`
+	}
+	cmd.RegisterCommand("boulder-foo", nil, &cmd.ConfigValidator{Config: &FooConfig{}, Validators: nil})
```

Comment:
This is what I meant when I said that having the two public Validate[JSON|YAML]ConfigForComponent functions take a validator as their argument, rather than a component name, would make unit-testing easier: if the component-name-to-validator lookup was handled by the calling code, then there would be no need to register a fake component in the global registry just for testing. It would also allow the unit tests to live in cmd/shell_test.go, because there wouldn't need to be a registry at all.

Reply:
I love this suggestion. Check out my refactor in the next couple commits.